src/base/bittorrent/torrenthandle.cpp

```diff
@@ -487,6 +487,7 @@ bool TorrentHandle::needSaveResumeData() const
 
 void TorrentHandle::saveResumeData()
 {
+    updateStatus(); // to update queue_position, see discussion in PR #6154
```

Comment:
Did I misunderstand you earlier? Every time when we generate resume info, we update status.